lib/parsers/ts-utils.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('node:fs');
+const ts = require('typescript');
+const { transformForLint } = require('./transform');
+const babel = require('@babel/core');
+const { replaceRange } = require('./transform');
+
+module.exports.patchTs = function patchTs() {
+  const readDirectory = ts.sys.readDirectory;
+  ts.sys.readDirectory = function (...args) {
+    const results = readDirectory.call(this, ...args);
+    return results.map((f) => f.replace(/\.gts$/, '.ts'));
+  };
+  ts.sys.fileExists = function (fileName) {
+    return fs.existsSync(fileName.replace(/\.mts$/, '.gts')) || fs.existsSync(fileName);
+  };
+  ts.sys.readFile = function (fname) {
+    let fileName = fname;
+    let content = '';
+    try {
+      content = fs.readFileSync(fileName).toString();
+    } catch {
+      fileName = fileName.replace(/\.mts$/, '.gts');
+      content = fs.readFileSync(fileName).toString();
```

Comment:
@patricklx don't we need to do transform here also since we are reading a `.gts` file?
And since filename is `.mts` we won't get inside next if condition